PKI/TLS improvements (#13622)
cockroachdb/docs: v22.1/security-reference/authentication.md

---
title: SQL Authentication
summary: An overview of Cluster Authentication Configuration capabilities and interface syntax
toc: true
docs_area: reference.security
---

> This page give an overview of CockroachDB's security features concerning authenticating the identity of SQL users attempting to connect with the cluster.

Reviewer: ```suggestion
This page give an overview of CockroachDB's security features for authenticating the identity of SQL users attempting to connect to the cluster.
```

Instead, you might be looking for:

> - [Logging in to the {{ site.data.products.db }} web console. ](../../cockroachcloud/authentication.html).

Reviewer: period and extra space in the link text.

- [Accessing the DB console on {{ site.data.products.core }} clusters](../ui-overview.html).

## Authentication configuration

> CockroachDB allows fine-grained configuration of which database connect attempts it allows to proceed to the authentication stage, and which authentication methods it will accept, based on:

Reviewer: ```suggestion
CockroachDB allows fine-grained configuration of which database connection attempts it allows to proceed to the authentication stage, and which authentication methods it accepts, based on:
```

- **Who** is making the attempt (SQL user).
- **Where** on the internet (IP Address) the attempt is coming from.

CockroachDB's authentication behavior is configured using a domain-specific language (DSL), shared with PostgreSQL, called host-based authentication (HBA).

> A specific CockroachDB cluster's authentication behavior is configured by setting its `server.host_based_authentication.configuration` [cluster setting](../cluster-settings.html), using the [`SET CLUSTER SETTING` statement](../set-cluster-setting.html), which accepts a single text field that must be a correctly formatted HBA manifest. Inspect the current setting with [`SHOW CLUSTER SETTING`.](../show-cluster-setting.html)

Reviewer: The first sentence seems like a bit of a run-on. Is there a link to a correctly formatted HBA manifest? I suggest something like "To inspect the current settings, use `SHOW CLUSTER SETTINGS`."

## Currently supported authentication methods

Authentication Method | CockroachDB Cloud | Supported in CockroachDB Core | CockroachDB Enterprise Support  
-------------|------------|-----|----
password              |      ✓              |           ✓                    |    ✓
[SCRAM-SHA-256](scram-authentication.html)         |      ✓              |           ✓                    |    ✓
certificate              |      &nbsp;         |           ✓                    |    ✓

> username/password combination              |      ✓              |           ✓                    |    ✓

Reviewer: Are we sure a password is required?

Author: not sure what you mean?
